tendrl/commons/message.py

```diff
@@ -0,0 +1,47 @@
+import datetime
+from inspect import getframeinfo
+from inspect import stack
+import uuid
+
+
+class Message(object):
```

Comment:
The Message class' interface should be as follows:

* Initialisation is done only when crafting a new message from scratch. This would be used by the entities that generate the messages. The initialise method should accept mandatory parameters directly, with perhaps a dict for the payload. Initialisation would thus automatically validate the presence of the mandatory parameters. This method should always generate a message_id itself.
* The class method `from_json` should be used when generating a message object from json. This would be the case in the node agent, when it reads a message from the socket. This method should raise an error clearly indicating that the json syntax is invalid, if applicable. This method must call the `validate` method (discussed below).
* The object should have a `to_json` method to dump the json format.
* The `validate` method should check for the existence of all the mandatory parameters, `message_id` among them. It must raise a validation error with the message indicating the precise problems.
* Whoever creates the message object using `from_json` would be responsible for catching the errors. In case of validation errors, it needs to initialise a new message object, whose payload would be the original invalid message object with the validation error messages. Eventually would be a good idea to create a way to properly handle nested objects, but this'll do for now.